pkg/proxy/iptables/proxier.go

```diff
@@ -1010,27 +1011,33 @@ func (proxier *Proxier) syncProxyRules() {
 
 		// Scan the endpoints list to see what we have. "hasEndpoints" will be true
 		// if there are any usable endpoints for this service anywhere in the cluster.
-		var hasEndpoints, hasLocalReadyEndpoints, hasLocalServingTerminatingEndpoints bool
+		var hasEndpoints, hasClusterReadyEndpoints, hasClusterServingTerminatingEndpoints, hasLocalReadyEndpoints, hasLocalServingTerminatingEndpoints bool
```

Comment:
This seems a bit messy, but should be cleaned up once we incorporate the changes from https://github.com/kubernetes/kubernetes/pull/106497, specifically by getting internal/external filtered endpoints from FilterEndpoints